numba/core/caching.py

```diff
@@ -555,7 +555,8 @@ def _open_for_write(self, filepath):
         Open *filepath* for writing in a race condition-free way (hopefully).
         uuid4 is used to try and avoid name collisions on a shared filesystem.
         """
-        tmpname = '%s.tmp.%d' % (filepath, uuid.uuid4())
+        uid = uuid.uuid4().hex[:16]  # avoid long paths
```

Comment:
```suggestion
        uid = hash(uuid.uuid4())  # avoid long paths
```
makes it shorter?

Reply:
not always:

```python

In [13]: uuid.uuid4().hex[:16]
Out[13]: '2814e0e5e9f24749'

In [14]: hash(uuid.uuid4())
Out[14]: 328160728412730801
```

Reply:
I guess I was lucky in my tiny sample size!
```python
In [24]: dd = defaultdict(int)

In [25]: for _ in range(1000000):dd[len(str(hash(uuid.uuid4())))]+=1

In [26]: dd
Out[26]: 
defaultdict(int,
            {19: 565845,
             18: 390817,
             16: 3887,
             17: 39052,
             15: 363,
             14: 32,
             13: 4})
```